### PR TITLE
Authorisation

### PR DESCRIPTION
**Pull request by kusiLaw**
### Milestone
- Install CanCanCan in your project.
- Add a role column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
  - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
  - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -22,6 +22,20 @@ def create
     end
   end
```
Comment (fobadara) [inline app/controllers/posts_controller.rb:26]: [REQUIRED] Kindly ensure CanCan is used for posts' authorization by adding this:

``` 
 class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing
✅ Good first attempt

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (kusiLaw) [reply app/controllers/posts_controller.rb:26]: Hi @fobadara,
thank you very much for your effort and the feedback. I really appreciate it and the changes will be added in my next commit. thank you.

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @kusiLaw, and @John-Kibirige👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Optional suggestions

- [ ] I would suggest adding your coding partner's info in the author section of the README file.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**